### PR TITLE
Update README with our Style Guidelines

### PR DESCRIPTION
**Pull request by joshuatbrown**

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -226,35 +223,36 @@ For UIKit view controllers, consider grouping lifecycle, custom accessors, and I
 
 ### Unused Code
 
-Unused (dead) code, including Xcode template code and placeholder comments should be removed. An exception is when your tutorial or book instructs the user to use the commented code.
+Unused (dead) code, including Xcode template code and placeholder comments should be removed.
 
-Aspirational methods not directly associated with the tutorial whose implementation simply calls the superclass should also be removed. This includes any empty/unused UIApplicationDelegate methods.
+Aspirational methods whose implementation simply calls the superclass should also be removed. This includes any empty or unused UIApplicationDelegate methods.
```
Comment (stevekohls) [inline README.markdown:228]:
```suggestion
Aspirational methods whose implementation simply calls the superclass should also be removed. This includes any empty or unused `UIApplicationDelegate` methods.
```

File: README.markdown [hunk 2]
```diff
@@ -112,7 +109,7 @@ For the above example using `UIGestureRecognizer`, 1 is unambiguous and preferre
 
 ### Class Prefixes
 
-Swift types are automatically namespaced by the module that contains them and you should not add a class prefix such as RW. If two names from different modules collide you can disambiguate by prefixing the type name with the module name. However, only specify the module name when there is possibility for confusion, which should be rare.
+Swift types are automatically namespaced by the module that contains them and you should not add a class prefix such as OB (for Oral-B). If two names from different modules collide you can disambiguate by prefixing the type name with the module name. However, only specify the module name when there is possibility for confusion, which should be rare.
```
Comment (stevekohls) [inline README.markdown:112]: Don't know if this should be in code font or not
```suggestion
Swift types are automatically namespaced by the module that contains them and you should not add a class prefix such as `OB` (for Oral-B). If two names from different modules collide you can disambiguate by prefixing the type name with the module name. However, only specify the module name when there is possibility for confusion, which should be rare.
```

File: README.markdown [hunk 3]
```diff
@@ -288,19 +286,17 @@ var deviceModels: [String]
...
 
 **Preferred**:
 ```swift
 if user.isHappy {
-  // Do something
-} else {
-  // Do something else
+    // Do something
+}
+else {
```
Comment (stevekohls) [inline README.markdown:298]: 😆

File: README.markdown [hunk 4]
```diff
@@ -309,36 +305,31 @@ if user.isHappy {
...
 ```swift
 class TestDatabase : Database {
-  var data :[String:CGFloat] = ["A" : 1.2, "B":3.2]
+    var data :[String:CGFloat] = ["A" : 1.2, "B":3.2]
 }
 ```
 
-* Long lines should be wrapped at around 70 characters. A hard limit is intentionally not specified.
-
+* Long lines should be wrapped at 125 characters. This makes GitHub code reviews easier, as the diff view shows a maximum of 125 characters.
```
Comment (stevekohls) [inline README.markdown:331]: Good idea about the GitHub code review line length. I didn't know that.
This is a good idea. Same held for 80 characters, back in the 80s 😄

File: README.markdown [hunk 5]
```diff
@@ -365,46 +356,48 @@ Here's an example of a well-styled class definition:
...
+        return "(\(x),\(y))"
+    }
 }
 ```
 
 The example above demonstrates the following style guidelines:
 
  + Specify types for properties, variables, constants, argument declarations and other statements with a space after the colon but not before, e.g. `x: Int`, and `Circle: Shape`.
- + Define multiple variables and structures on a single line if they share a common purpose / context.
+ + Define a single variable or structure on a single line.
```
Comment (stevekohls) [inline README.markdown:400]: 👍🏻

File: README.markdown [hunk 6]
```diff
@@ -457,20 +449,18 @@ Keep short function declarations on one line including the opening brace:
 
 ```swift
 func reticulateSplines(spline: [Double]) -> Bool {
-  // reticulate code goes here
+    // reticulate code goes here
 }
 ```
 
-For functions with long signatures, put each parameter on a new line and add an extra indent on subsequent lines:
+For functions with long signatures, put the first parameter on the same line, and subsequent parameters new lines, ensuring that parameter names are left-aligned:
```
Comment (stevekohls) [inline README.markdown:456]: 👍🏻

File: README.markdown [hunk 7]
```diff
@@ -558,9 +547,9 @@ Chained methods using trailing closures should be clear and easy to read in cont
 let value = numbers.map { $0 * 2 }.filter { $0 % 3 == 0 }.index(of: 90)
 
 let value = numbers
```
Comment (stevekohls) [inline README.markdown:549]: I typically chain with the first call inline, like with parameters in functions. But I tend not to do lots of chained operations like this. Instead using intermediate results for readability and understanding.

Also, there should be spaces inside the `{` `}` like this: `{ $0 * 2 }`
Thoughts?

```suggestion
let value = numbers.map { $0 * 2 }
...
```

File: README.markdown [hunk 8]
```diff
@@ -764,25 +752,25 @@ Free functions are most appropriate when they aren't associated with any particu
...
-let tuples = zip(a, b)  // feels natural as a free function (symmetry)
-let value = max(x, y, z)  // another free function that feels natural
+let tuples = zip(a, b) // feels natural as a free function (symmetry)
+let value = max(x, y, z) // another free function that feels natural
 ```
 
 ## Memory Management
 
-Code (even non-production, tutorial demo code) should not create reference cycles. Analyze your object graph and prevent strong cycles with `weak` and `unowned` references. Alternatively, use value types (`struct`, `enum`) to prevent cycles altogether.
+Code should not create reference cycles. Analyze your object graph and prevent strong cycles with `weak` and `unowned` references. Alternatively, use value types (`struct`, `enum`) to prevent cycles altogether.
```
Comment (stevekohls) [inline README.markdown:773]: I think we should drop this bit:
> Alternatively, use value types (`struct`, `enum`) to prevent cycles altogether.

There's lots of ways to prevent retain cycles, I'd rather not call out just one.

File: README.markdown [hunk 9]
```diff
@@ -1025,48 +1014,42 @@ When building a long string literal, you're encouraged to use the multi-line str
...
-Likewise, do not use Xcode's ability to drag a color or an image into a source statement. These turn into #colorLiteral and #imageLiteral, respectively, and present unpleasant challenges for a reader trying to enter them based on tutorial text. Instead, use `UIColor(red:green:blue)` and `UIImage(imageLiteralResourceName:)`.
-
-## Organization and Bundle Identifier
-
-Where an Xcode project is involved, the organization should be set to `Ray Wenderlich` and the Bundle Identifier set to `com.raywenderlich.TutorialName` where `TutorialName` is the name of the tutorial project.
-
-![Xcode Project settings](screens/project_settings.png)
+Likewise, do not use Xcode's ability to drag a color or an image into a source statement. These turn into #colorLiteral and #imageLiteral, respectively, and present unpleasant challenges for the reader. Instead, use `UIColor(red:green:blue)` and `UIImage(imageLiteralResourceName:)`.
 
 ## Copyright Statement
```
Comment (stevekohls) [inline README.markdown:1054]: We will need to change the copyright, but that can be a separate PR.

OR, should we remove this from the standards?

File: README.markdown [hunk 10]
```diff
@@ -226,35 +223,36 @@ For UIKit view controllers, consider grouping lifecycle, custom accessors, and I
...
-  return Database.contacts.count
+    // #warning Incomplete implementation, return the number of rows
+    return Database.contacts.count
 }
 
 ```
+
 ### Minimal Imports
 
 Import only the modules a source file requires. For example, don't import `UIKit` when importing `Foundation` will suffice. Likewise, don't import `Foundation` if you must import `UIKit`.
```
Comment (stevekohls) [inline README.markdown:258]: This examples in this section don't have `swift` after the "```", so they're not color coding properly. Might be worth a search to see if there's others.

File: README.markdown [hunk 11]
```diff
@@ -288,19 +286,17 @@ var deviceModels: [String]
 
 ## Spacing
 
-* Indent using 2 spaces rather than tabs to conserve space and help prevent line wrapping. Be sure to set this preference in Xcode and in the Project settings as shown below:
-
-![Xcode indent settings](screens/indentation.png)
-
+* Indent using 4 spaces rather than tabs. Be sure to set this preference in Xcode and in the Project settings.
 * Method braces and other braces (`if`/`else`/`switch`/`while` etc.) always open on the same line as the statement but close on a new line.
```
Comment (stevekohls) [inline README.markdown:290]: I made a comment elsewhere about closures using the shorthand parameter notation `$1`, `$2`, etc. We should have a note about putting spaces after opening and before closing braces in all cases.
```swift
    ...
    numbers.map { $0 * 2 }
    ...
```

**Review by stevekohls — CHANGES_REQUESTED** [review]
@joshuatbrown Looks really good! Some minor comments. Let me know what you think about my questions.

File: README.markdown (hunk 8 above)
Comment (joshuatbrown) [reply README.markdown]: Agreed, and done. For the review process: do I mark this as resolved now, since I did it, or do you mark it as resolved once you've confirmed I did it the way you intended?

File: README.markdown (hunk 2 above)
Comment (joshuatbrown) [reply README.markdown:108]: I'll take it. 👍

File: README.markdown (hunk 10 above)
Comment (joshuatbrown) [reply README.markdown:254]: Done!

File: README.markdown (hunk 9 above)
Comment (joshuatbrown) [reply README.markdown]: I removed it after your review. I was hopeful we wouldn't need a copyright notice in each file, but if we do, I suppose including it here makes sense.

File: README.markdown (hunk 7 above)
Comment (joshuatbrown) [reply README.markdown:531]: I don't often chain operations like this, either, so I wonder if we should remove this section or make the update you suggest...

Comment (stevekohls) [reply README.markdown:531]: Not sure? as long as they're on multiple lines, I'm good. If something is unclear, that's where I'd prefer breaking it up into separate chunks.
Maybe the preference of making chaining clear and easy to read is good, just the formatting is the issue.

File: README.markdown [hunk 12]
```diff
@@ -429,20 +425,6 @@ var diameter: Double {
 }
 ```
 
-### Final
```
Comment (stevekohls) [inline README.markdown:440]: And the only time I've seen a worthwhile use for `final` is in writing a library that we don't want people to extend

File: README.markdown (hunk 11 above)
Comment (joshuatbrown) [reply README.markdown:286]: Adding a little here and a little there, but open to suggestions on how to do it better.
